WordPress/gutenberg
# Block Support: Add text transform block support using CSS variables

## Description

This adds block support for text transform styles and is a continuation from discussion around https://github.com/WordPress/gutenberg/pull/25641 around adding font style block support.

#### Changes Included
* Adds block support for text transforms
* Allows for preset text transform options and uses CSS variables
* Updates heading block to opt-in to text transform support for easier testing

#### Notes
* The UI for the text transform control will need some design input. It fills the need for testing this now though. 
* The buttons within the controls simply use plain text ( e.g 'AB', 'ab', 'Ab' ) at the moment and still need icons.
* The intention is when this and text decoration block support (https://github.com/WordPress/gutenberg/pull/26059) are both merged a quick update will be made to include both sets of controls so they display on a single row within the typography controls.

## How has this been tested?
Manually tested using heading block.

#### Test Instructions

1. Add a heading block to a post and select it.
2. You should see new text transform buttons under the Inspector Controls > Typography section.
3. Toggle the various text transform buttons and confirm the block updates visually as you'd expect.
4. With a text transform toggled on, inspect the heading block in dev tools and confirm that the block includes an inline style using `var()` and an appropriate CSS variable.
5. Save the post and view on the frontend.
6. The same text transform choice should be reflected on the frontend block's inline styles.

## Screenshots 
![TextTransformUI](https://user-images.githubusercontent.com/60436221/96689932-655d5580-13c6-11eb-89fe-d4c12b0ae0df.gif)


## Types of changes
Enhancement

## Next Steps
- [ ] Update tests in `class-block-supported-styles-test.php` if needed after approach confirmed.
- [ ] Add text transforms to the Global styles Typography panel once UI is refined.

## Checklist:
